test/contrib/automotive/obd/scanner.uts

```diff
@@ -140,111 +135,111 @@ from subprocess import call
+s8_tid00 = OBD() / OBD_S08_PR(data_records=[OBD_S08_PR_Record() / OBD_TID00(supported_tids="")])
+s9_iid00 = OBD() / OBD_S09_PR(data_records=[OBD_S09_PR_Record() / OBD_IID00(supported_iids="")])
+
+
 s1_pid01 = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID01()])
 s1_pid03 = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID03(fuel_system1=0, fuel_system2=2)])
 s1_pid0B = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID0B(data=100)])
 s1_pid0F = OBD()/OBD_S01_PR(data_records=[OBD_S01_PR_Record()/OBD_PID0F(data=50)])
-s1_queue = queue.Queue()
-s1_queue.put(s1_pid00)
-s1_queue.put(s1_pid00)
-s1_queue.put(s1_pid03)
-s1_queue.put(s1_pid0B)
-s1_queue.put(s1_pid0F)
-s1_queue.put(s1_pid01)
 
-s3 = OBD()/OBD_S03_PR(dtcs=[OBD_DTC()])
-s3_queue = queue.Queue()
-s3_queue.put(s3)
+example_responses = \
+    [ECUResponse(session=range(0, 255), security_level=0, responses=s3),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s6_mid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s8_tid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s9_iid00),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid01),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid03),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid0B),
+     ECUResponse(session=range(0, 255), security_level=0, responses=s1_pid0F)]
 
-+ Simulate scanner
 
-= Create responder
-exit_if_no_isotp_module()
-# Ensures the responder is running before sending the first request
-ready = Event()
-
-class MockResponder(Thread):
-    daemon = True
-    def __init__(self):
-        Thread.__init__(self)
-        self._stopped = Event()
-    def process_request(self, req, sock):
-        if req.service == 1:
-            resp = s1_queue.get()
-            sock.send(resp)
-        elif req.service == 3:
-            resp = s3_queue.get()
-            sock.send(resp)
-    def run(self):
-        with ISOTPSocket(new_can_socket(iface0), 0x7e8, 0x7e0, basecls=OBD, padding=True) as sock:
-            while not self._stopped.is_set() or not (s1_queue.empty() and s3_queue.empty()):
-                sock.sniff(timeout=5, started_callback=ready.set, store=False, prn=lambda p: self.process_request(p, sock))
-    def stop(self):
-        self._stopped.set()
-
-responder = MockResponder()
-responder.start()
-
-= Get ids
-exit_if_no_isotp_module()
-drain_bus(iface0)
-with ISOTPSocket(new_can_socket(iface0), 0x7e0, 0x7e8, basecls=OBD, padding=True) as socket:
-    ready.wait(timeout=5)
-    all_ids_set = set(range(1, 256))
-    supported_ids = _supported_id_numbers(socket, 0.1, OBD_S01, 'pid', False)
-    unsupported_ids = all_ids_set - supported_ids
++ Simulate scanner
 
 = Run scanner
+
 exit_if_no_isotp_module()
```

Comment:
You should be able to remove this call. The initial setup should work on every Travis system, and on Windows.